Issue#9 static syntax checkers

VictorOsterman opened the pull request:
closes #9

VictorOsterman committed:
add: pyright version in requirement.txt

VictorOsterman committed:
fix: import related to issue with git.exc.GitCommandError

VictorOsterman committed:
fix: issues related to dummy.py and test_dummy.py

VictorOsterman committed:
fix: issue connected to import

VictorOsterman committed:
refactor: move run_test.py to utils to resolve issue where import from tests occurred

VictorOsterman committed:
add: functionality to run pyright on .py files in src/ and tests/ dir

VictorOsterman committed:
remove: main in run_syntax file

VictorOsterman committed:
fix: run_syntax file in a manner that it can be handled correctly by run_build

VictorOsterman committed:
move: run_syntax to utils directory

VictorOsterman committed:
remove: pyrightconfig.json file

Since a path towards tmp (which can be accessed from any directory), the pyrightconfig file isn't required any more

VictorOsterman committed:
Merge branch 'assessment' into issue/9-Static-Syntax-Checkers

Tobiky reviewed (changes requested):
Needs some tests, make one or two tests that executes the syntax checker function on some test files.

VictorOsterman committed:
add: test for syntax_checker

first test: check if pyright doesn't find any error when it shouldn't

second test: check if pyright find an error when it should (missing + between a and b)

VictorOsterman commented:
> Needs some tests, make one or two tests that executes the syntax checker function on some test files.

You are correct, my bad. Now, I've implemented two tests. 
The first checks if pyright:s return code is zero when the code is code. 
The second test checks that pyright actually finds a syntax error in the code (missing '+' between a and b in the add method)

Tobiky reviewed (approved):
Nova, good job. Feel free to merge at your convenience.
